flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/executor/BatchExecutor.java

```diff
@@ -42,47 +43,108 @@
-
-		// TODO supports streamEnv.execute(streamGraph)
-		try {
-			return execEnv.execute(getNonEmptyJobName(jobName));
-		} finally {
-			transformations.clear();
-		}
+		StreamGraph streamGraph = generateStreamGraph(transformations, jobName);
+		return execEnv.execute(streamGraph);
 	}
 
-	public static StreamGraph generateStreamGraph(
-		StreamExecutionEnvironment execEnv,
-		List<Transformation<?>> transformations,
-		String jobName) throws Exception {
-		// TODO avoid cloning ExecutionConfig
-		ExecutionConfig executionConfig = InstantiationUtil.clone(execEnv.getConfig());
+	/**
+	 * Backup previous streamEnv config and set batch configs.
+	 */
+	private void backupAndUpdateStreamEnv(StreamExecutionEnvironment execEnv) {
+		ExecutionConfig executionConfig = execEnv.getConfig();
+
+		enableObjectReuse = executionConfig.isObjectReuseEnabled();
 		executionConfig.enableObjectReuse();
+
+		latencyTrackingInterval = executionConfig.getLatencyTrackingInterval();
 		executionConfig.setLatencyTrackingInterval(-1);
 
-		return new StreamGraphGenerator(transformations, executionConfig, new CheckpointConfig())
-			.setChaining(execEnv.isChainingEnabled())
-			.setStateBackend(execEnv.getStateBackend())
-			.setDefaultBufferTimeout(-1)
-			.setTimeCharacteristic(TimeCharacteristic.ProcessingTime)
-			.setUserArtifacts(execEnv.getCachedFiles())
-			.setSlotSharingEnabled(false)
-			.setScheduleMode(ScheduleMode.LAZY_FROM_SOURCES)
-			.setJobName(jobName)
-			.generate();
+		timeCharacteristic = execEnv.getStreamTimeCharacteristic();
+		execEnv.setStreamTimeCharacteristic(TimeCharacteristic.ProcessingTime);
+
+		bufferTimeout = execEnv.getBufferTimeout();
+		execEnv.setBufferTimeout(-1);
+
+		inputDependencyConstraint = executionConfig.getDefaultInputDependencyConstraint();
+		if (isShuffleModeAllBatch()) {
+			executionConfig.setDefaultInputDependencyConstraint(InputDependencyConstraint.ALL);
+		}
 	}
 
+	/**
+	 * Restore previous streamEnv after execute batch jobs.
+	 */
+	private void restoreStreamEnv(StreamExecutionEnvironment execEnv) {
+		ExecutionConfig executionConfig = execEnv.getConfig();
+		if (enableObjectReuse) {
+			executionConfig.enableObjectReuse();
+		} else {
+			executionConfig.disableObjectReuse();
+		}
+		executionConfig.setLatencyTrackingInterval(latencyTrackingInterval);
+		execEnv.setStreamTimeCharacteristic(timeCharacteristic);
+		execEnv.setBufferTimeout(bufferTimeout);
+		if (isShuffleModeAllBatch()) {
+			executionConfig.setDefaultInputDependencyConstraint(inputDependencyConstraint);
+		}
+	}
+
+	/**
+	 * Translates transformationList to streamGraph.
+	 */
+	public StreamGraph generateStreamGraph(
+			List<Transformation<?>> transformations,
+			String jobName) throws Exception {
+		StreamExecutionEnvironment execEnv = getExecutionEnvironment();
+		backupAndUpdateStreamEnv(execEnv);
+		transformations.forEach(execEnv::addOperator);
```

Comment:
If Exception is thrown here, you will not restore execEnv

Reply:
Now I restore execEnv at last and remove throwing exception.